### PR TITLE
Ensure external threads always get a Logger and FileResolver

### PR DESCRIPTION
**Pull request by dvicini**
This PR ensures that any external thread that is registered with Mitsuba will have a valid logger and file resolver.

The Logger is re-used from the main thread, since it is thread safe. The file resolver is copied, as it is not thread safe, but light weight to copy.

This PR also adds two tests that exercise this newly added / fixed functionality.

### CODE REVIEW

File: src/core/python/logger.cpp [hunk 1]
```diff
@@ -29,18 +35,6 @@ static void PyLog(mitsuba::LogLevel level, const std::string &msg) {
     if (!name.empty() && name[0] != '<')
         fmt.insert(2, "()");
 
-    if (!Thread::thread()->logger()) {
```
Comment (merlinND) [inline src/core/python/logger.cpp:32]: Maybe this could be replaced with an assert?

Comment (dvicini) [reply src/core/python/logger.cpp:32]: I am not against that. My goal was to make it consistent with the C++ side here, which also just ignores this case:
https://github.com/mitsuba-renderer/mitsuba3/blob/master/include/mitsuba/core/logger.h#L140

Comment (merlinND) [reply src/core/python/logger.cpp:32]: Understandable.. I guess it will crash  either way :D

Comment (dvicini) [reply src/core/python/logger.cpp:32]: It should just ignore and not log anything at all, which is not too bad.

Comment (merlinND) [reply src/core/python/logger.cpp:32]: Ahh sorry I didn't pay attention to the check you added above. Makes sense.

The case of `!Thread::thread()->logger()` should no longer happen by default now that you automatically bring over the main thread's logger, right?
If there's no logger, then it must be that the user set a thread environment that explicitly doesn't have a logger?

Comment (dvicini) [reply src/core/python/logger.cpp:32]: Yes exactly. Not sure if that's something one would ever want?

In any case, there are still some other issues to be fixed with this PR. The current version does not shut down cleanly..

File: src/core/tests/test_thread.py [hunk 2]
```diff
@@ -3,11 +3,41 @@
...
     thread.join()
+
+def test02_access_logger_from_new_thread(variant_scalar_rgb):
+    def print_to_log():
+        mi.set_log_level(mi.LogLevel.Info)
+        mi.Log(mi.LogLevel.Info, "Log from a thread environment.")
+
+    thread = Thread(target = print_to_log)
+    thread.start()
+    thread.join()
```
Comment (njroussel) [inline src/core/tests/test_thread.py:22]: Let's add an assert to make sure that the log was visible (you can captue it with the `capsys` fixture).

**Review by njroussel — APPROVED** [review]
One tiny request, the rest looks good to me!

I've checked all of the build logs, none of the environments have leaks when running the test suite :+1:

File: src/core/tests/test_thread.py (hunk 2 above)
Comment (dvicini) [reply src/core/tests/test_thread.py:34]: Good idea. I now added a `mi.StreamAppender`. I could not get it to work with `capsys`, this one only would capture Python-side prints for me (seems to be a known limitation)

**Pull request merged**